app/src/main/java/de/tum/in/tumcampusapp/component/tumui/calendar/CreateEventActivity.kt

```diff
@@ -86,16 +95,46 @@ class CreateEventActivity : ActivityForAccessingTumOnline<CreateEventResponse>(R
         setDateAndTimeListeners()
 
         createEventButton.setOnClickListener {
-            if (end.isBefore(start)) {
+            if (end.isBefore(start) || (repeats && last.isBefore(start))) {
                 showErrorDialog(getString(R.string.create_event_time_error))
                 return@setOnClickListener
             }
+            if (repeats) {
+                if (endAfterRadioBtn.isChecked && (eventRepeatsTimes.text.toString() == "" ||
+                        eventRepeatsTimes.text.toString().toInt() < 2)) {
+                    showErrorDialog(getString(R.string.create_event_too_little_error))
+                    return@setOnClickListener
+                }
+                // Don't allow too many requests
+                if ((endOnRadioBtn.isChecked && start.plusMonths(6).isBefore(last)) ||
+                        (endAfterRadioBtn.isChecked && eventRepeatsTimes.text.toString().toInt() > 25)) {
+                    showErrorDialog(getString(R.string.create_event_too_many_error))
+                    return@setOnClickListener
+                }
+            }
             if (isEditing) {
                 editEvent()
             } else {
                 createEvent()
             }
         }
+
+        // edited events cannot repeat
+        if (isEditing) {
+            repeatingSwitch.layoutParams = LinearLayout.LayoutParams(0, 0)
+        } else {
+            repeatingSwitch.setOnCheckedChangeListener { _, isChecked ->
+                if (isChecked) {
+                    repeats = true
```

Comment:
nit: To be honest: this value should be read in the `createEvent` function if possible. All the data that is stored on the class level makes it hard to debug in a failure scenario, because there are so many paths to get to a certain state. (Feel free to ignore this - its just me being unhappy with android)